newIDE/app/src/ResourcesList/ResourceThumbnail/ImageThumbnail.js

```diff
@@ -48,6 +56,15 @@ const ImageThumbnail = ({
   onContextMenu,
   muiTheme,
 }) => {
+  let warningSize = false;
+
+  const img = new Image();
+  img.src = resourcesLoader.getResourceFullUrl(project, resourceName);
+
+  if (img.width > 2048 || img.height > 2048) {
+    warningSize = true;
+  }
```

Comment:
Could you do this as a callback to the `<img>` element that is below?
`onLoad` prop should be called with the `HTMLImageElement` as parameter, on which you can read width/height. At this time, prefer to set the state (use a hook: `const [hasSizeWarning, setHasWarningSize] = React.useState(false);` then `setHasWarningSize(true);` instead of `warningSize = true`).

Reply:
Thank you it's better :)